### ISSUE TITLE
Recent change is now hanging our build

### ISSUE THREAD

**Issue by justinmchase** [vgamula/gulp-cucumber#33]
Just an FYI for anyone who is experiencing hangs in their build after upgrading to `gulp-connect` `v0.0.19`

The change in PR #29 was a breaking change. We had some logic that was listening for the `'end'` event and was then shutting down our test servers.

Its fixable by changing your logic from something like:

```
gulp.task('test:client:integration', function (callback) {
    var options = { ... };
    gulp
        .src(paths.clients.features)
        .pipe($.cucumber(options)).on('end', function () {
          testServer.shutdown(callback)
        });
});
```

to:
```
gulp.task('test:client:integration', ['test:client:cucumber'], function (callback) {
    testServer.shutdown(callback);
});
gulp.task('test:client:cucumber', function () {
    var options = { ... };
    return gulp 
        .src(paths.clients.features)
        .pipe($.cucumber(options));
});
```

**Comment by justinmchase** [comment]
It turns out the simple change I mentioned above is not quite right. Gulp should actually be emitting an 'end' event in non-error cases.

**Comment by vgamula** [comment]
Fixed in v0.0.20.
Feel free to reopen this issues if there will be any problems.
